chef/cookbooks/bind9/recipes/default.rb

```diff
@@ -404,7 +407,10 @@ def make_zone(zone)
             allow_transfer: allow_transfer,
             ipaddress: admin_addr,
```

Comment:
simply change ipaddress here to public_addr if enable_designate is added? then you don't need to do the if/else inside the template?

Reply:
That's good for designate, but this is the DNS infrastructure for the crowbar deployed cloud. If we make named/DNS _only_ listen on the public network then any node in the cloud that isn't on the public network can't access DNS. So unless we also go and make sure _every_ node can route to the DNS public ips then there will be nodes in the cloud that can't talk to DNS. Which I think is more problems then an if in a template.

ATM every node in the cloud, all compute, controllers, storage nodes listen on the admin network. Which means they can talk to the crowbar DNS infra, but not all listen on the Public network.

that's my 2 cents anyway.